parsing/builtin_attributes.ml

```diff
@@ -36,13 +36,22 @@ let attr_order a1 a2 =
   | 0 -> Int.compare a1.loc.loc_start.pos_cnum a2.loc.loc_start.pos_cnum
   | n -> n
 
+let compiler_stops_before_attributes_consumed () =
+  let stops_before_lambda =
+    match !Clflags.stop_after with
+    | None -> false
+    | Some pass -> Clflags.Compiler_pass.(compare pass Lambda) < 0
+  in
+  stops_before_lambda || !Clflags.print_types
```

Comment:
I would propose to update the documentation of `warn_unused` to mention that the warnings are only emitted if the compilation is not stopped early.

Reply:
Good point! Added.